daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilParseOutputStreamContentHandler.scala

```diff
@@ -67,12 +73,64 @@ class DaffodilParseOutputStreamContentHandler(out: OutputStream, pretty: Boolean
   override def startPrefixMapping(prefix: String, uri: String): Unit = {
     val _prefix = if (prefix == "") null else prefix
     prefixMapping = NamespaceBinding(_prefix, uri, prefixMapping)
+    currentElementPrefixMapping = NamespaceBinding(_prefix, uri, currentElementPrefixMapping)
   }
 
   override def endPrefixMapping(prefix: String): Unit = {
     // do nothing
   }
 
+  /**
+   * Uses Attributes, which is passed in to the startElement callback, to extract prefix mappings
+   * and other attributes and return a tuple of an updated prefixMapping and a Seq of attribute=val
+   * String pairings
+   */
+  def processAttributes(atts: Attributes, prefixMapping: NamespaceBinding): (NamespaceBinding, Seq[String]) = {
+    var pm = prefixMapping
+    var attrPairings: Seq[String] = Seq()
+    var i = 0
+    while (i < atts.getLength) {
+      val qName = atts.getQName(i)
+      try {
+        val pre = qName.replace("xmlns:", "")
+        pm.getURI(pre)
+        // we make this call to check if the prefix already exists. If it doesn't exist, we get the
+        // npe so we can add it, but if it does exist, nothing happens and it doesn't get readded
+        // and we instead proceed to the next item in Attributes
+      } catch {
+        // prefix does not exist, so lookup resulted in npe, therefore
+        // proceed with scraping Attributes
+        case _: NullPointerException => {
+          val attrVal =  atts.getValue(i)
+          if (qName == "xmlns") {
+            pm = NamespaceBinding(null, attrVal, pm)
+          } else if (qName.startsWith("xmlns:")) {
+            val prefix = qName.substring(6)
+            pm = NamespaceBinding(prefix, attrVal, pm)
+          } else {
+            // if not a namespace mapping, it is an attribute we're interested in
+            if (qName.nonEmpty) {
+              attrPairings :+= s" $qName=" + "\"" + s"${attrVal}" + "\""
```

Comment:
We shoudl avoid this seq append, perhaps do a prepend and then reverse at the end if order matters.

Also, this might be cleaner using tripple quotes, e.g.
```scala
s""" ${qName}="${attrVal}""""
```